src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_ssl.c

```diff
@@ -354,13 +354,24 @@ PAL_TlsIo AppleCryptoNative_SslRead(SSLContextRef sslContext, uint8_t* buf, uint
     return OSStatusToPAL_TlsIo(status);
 }
 
-int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringRef cfHostname, CFDateRef notBefore)
+int32_t AppleCryptoNative_SslIsHostnameMatch(SSLContextRef sslContext, CFStringRef cfHostname, CFDateRef notBefore, int32_t* pOSStatus)
 {
+    if (pOSStatus == NULL)
+        return -1;
+
     if (sslContext == NULL || notBefore == NULL)
+    {
+        *pOSStatus = errSecInvalidData;
         return -1;
+    }
+
     if (cfHostname == NULL)
+    {
+        *pOSStatus = errSecInvalidData;
         return -2;
+    }
 
+    *pOSStatus = noErr;
     SecPolicyRef sslPolicy = SecPolicyCreateSSL(true, cfHostname);
 
     if (sslPolicy == NULL)
```

Comment:
We don't need to set *pOSStatus to an error code if sslPolicy or certs is null?

Reply:
I did not find any OSSstatus value I liked. I'l probably change trace log and I'll log the negative return value.